wob.c

```diff
@@ -381,13 +390,13 @@ main(int argc, char **argv)
 					break;
 				}
 
-				fgets_rv = fgets(input_buffer, 6, stdin);
+				fgets_rv = fgets(input_buffer, 7, stdin);
 				if (feof(stdin)) {
 					wob_destroy(&app);
 					return EXIT_SUCCESS;
 				}
 
-				if (fgets_rv == NULL || !wob_parse_input(input_buffer, &percentage)) {
+				if (fgets_rv == NULL || !wob_parse_input(input_buffer, &percentage, &maximum)) {
```

Comment:
I think, it would be better to move "if percentage > maximum" validation outside `wob_parse_input` function. It's much simpler this way :)

By the way, `maximum` variable shouldn't be passed as pointer - it doesn't make much sense in this context as it is read only.

```suggestion
				if (fgets_rv == NULL || !wob_parse_input(input_buffer, &percentage) || percentace > maximum) {
```

Reply:
You're right about that. A bool argument could be added to wob_parse_input, or just have another if condition after the if statement at line 399-403, however it might be clearer to let wob_parse_input handle the parsing instead of doing it outside that function.